File: python/leetcode/53. Maximum Subarray.py
class Solution:
    def maxSubArray(self, nums: List[int]) -> int:
        if len(nums) == 0:
            return 0
        max = nums[0]
        sum = max

        for i in range(1, len(nums)):
            sum = nums[i] if nums[i] + sum < nums[i] else nums[i] + sum
            max = max if max > sum else sum
        return max

File: go/leetcode/53. Maximum Subarray.go
func maxSubArray(nums []int) int {
    if len(nums) < 1 {return 0}
    
    max := nums[0]
    sum := max
    
    for i := 1; i < len(nums); i++ {
        if nums[i] > nums[i] + sum {
            sum = nums[i]
        } else {sum = nums[i] + sum}
        if max < sum {
            max = sum
        }
    }
    
    return max
    
} 